server/service/teams_test.go

```diff
@@ -371,6 +371,58 @@ func TestApplyTeamSpecs(t *testing.T) {
 	})
 }
 
+// Tests that a new enroll secret is created for new teams when none are provided
+func TestApplyTeamSpecEnrollSecretForNewTeams(t *testing.T) {
```

Comment:
Do you know if we have test coverage for how enroll secrets are handled in the edit team from spec scenarios? Just to confirm that these changes didn't unexpectedly change that behavior? Based on the PR, I think the chance of regression is very, very low so I'm asking more just for future peace of mind.